enterprise/tailnet/workspaceproxy.go

```diff
@@ -0,0 +1,98 @@
+package tailnet
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/enterprise/wsproxy/wsproxysdk"
+	agpl "github.com/coder/coder/tailnet"
+)
+
+func ServeWorkspaceProxy(conn net.Conn, ma agpl.MultiAgentConn) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		err := forwardNodesToWorkspaceProxy(ctx, conn, ma)
+		if err != nil {
+			cancel()
```

Comment:
This doesn't seem to do anything, should it close the conn?

Reply:
Good point